### ISSUE TITLE
False Positives for Multiple Packages (CVE-2018-0786)

### ISSUE THREAD

**Issue by thefringeninja** [RetireNet/dotnet-retire#41]
According to dotnet/corefx#26236, multiple `System.ServiceModel.*` packages were vulnerable at 4.4.0 but this is fixed in 4.4.1. However this tool is still reporting them as vulnerabilities. Output of `dotnet retire loglevel=debug`:
```
dbug: dotnet_retire.RetireLogger[0]
      Found in total 60 references of NuGets (direct & transient)
fail: dotnet_retire.RetireLogger[0]
      Found use of 6 vulnerable libs in 36 dependency paths.
      
      * System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Duplex/4.4.1
            ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Duplex/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Http/4.4.1
            ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Http/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.NetTcp/4.4.1
            ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.NetTcp/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Primitives/4.4.1
            ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Primitives/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Duplex/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Duplex/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
              ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Http/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Http/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
              ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.NetTcp/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.NetTcp/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
              ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Security/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Security/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
              ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Security/4.4.1
            ╚ System.Private.ServiceModel/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Security/4.4.1
             ╚ System.Private.ServiceModel/4.4.1
      
      * System.ServiceModel.Duplex/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Duplex/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Duplex/4.4.1
      
      * System.ServiceModel.Http/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Http/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Http/4.4.1
      
      * System.ServiceModel.NetTcp/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.NetTcp/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.NetTcp/4.4.1
      
      * System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Duplex/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Duplex/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Http/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Http/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.NetTcp/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.NetTcp/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Security/4.4.1
            ╚ System.ServiceModel.Primitives/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Security/4.4.1
             ╚ System.ServiceModel.Primitives/4.4.1
      
      * System.ServiceModel.Security/4.4.1
      
          Topshelf/4.1.0
          ╚ Microsoft.Windows.Compatibility/2.0.1
           ╚ System.ServiceModel.Security/4.4.1
      
          Topshelf/4.1.0
          ╚ TopShelf.ServiceInstaller/4.1.0
           ╚ Microsoft.Windows.Compatibility/2.0.1
            ╚ System.ServiceModel.Security/4.4.1
      
info: dotnet_retire.RetireLogger[0]
      Scan complete.
```

**Comment by johnkors** [comment]
Yes, 4.4.0 was reported as a vulnerability in January, with a fix in 4.4.1. However, later on (june), 4.4.1 was reported to have issues as well in https://github.com/dotnet/announcements/issues/73 .

**Comment by thefringeninja** [comment]
Awesome, working as intended then.
